quotetracker/coinmarketcap.go

```diff
@@ -2,35 +2,60 @@ package quotetracker
 
 import (
 	"context"
-	"encoding/json"
 	"errors"
-	"io/ioutil"
 	"net/http"
 	"net/url"
 	"time"
 )
 
 const coinMarketCapURL = "https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/latest"
 
-type apiData struct {
-	Data   map[string]*curData `json:"data"`
-	Status status              `json:"status"`
+type coinMarketCapResponse struct {
+	pair Pair
+
+	Data   map[string]*coinMarketCapData `json:"data"`
+	Status coinMarketCapStatus           `json:"status"`
 }
 
-type status struct {
+type coinMarketCapStatus struct {
 	ErrorMessage string    `json:"error_message"`
 	Timestamp    time.Time `json:"timestamp"`
 }
 
-type curData struct {
-	Quote map[string]*quoteData `json:"quote"`
+type coinMarketCapData struct {
+	Quote map[string]*coinMarketCapQuote `json:"quote"`
 }
 
-type quoteData struct {
+type coinMarketCapQuote struct {
 	Price     float64 `json:price`
 	Volume24h float64 `json:volume_24h`
```

Comment:
should these be quoted?
```suggestion
type coinMarketCapQuote struct {
	Price     float64 `json:"price"`
	Volume24h float64 `json:"volume_24h"`
}
```

Reply:
oops. yes

Reply:
37dc827